DiagnosticsExtension/Models/ConnectionStringValidator/EventHubsValidator.cs

```diff
@@ -51,53 +53,110 @@ async public Task<ConnectionStringValidationResult> ValidateAsync(string connect
+            string eventHubName = entityName;
+
+            if (envDict.Contains(appSettingName))
+            {
+                try
                 {
-                    response.Status = ConnectionStringValidationResult.ResultStatus.DnsLookupFailed;
+                    string connectionString = Environment.GetEnvironmentVariable(appSettingName);
+                    client = new EventHubProducerClient(connectionString);
                 }
-                else if (e.Message.Contains("InvalidSignature"))
+                catch (Exception e)
                 {
-                    response.Status = ConnectionStringValidationResult.ResultStatus.AuthFailure;
+                    throw new MalformedConnectionStringException(e.Message, e);
                 }
-                else if (e.Message.Contains("Ip has been prevented to connect to the endpoint"))
+            }
+            else
+            {
+                try
                 {
-                    response.Status = ConnectionStringValidationResult.ResultStatus.Forbidden;
+                    string serviceUriString = Environment.GetEnvironmentVariable(appSettingName + ConnectionStringResponseUtility.FullyQualifiedNamespace);
+                    appSettingClientIdValue = Environment.GetEnvironmentVariable(appSettingName + ConnectionStringResponseUtility.ClientId);
+                    appSettingClientCredValue = Environment.GetEnvironmentVariable(appSettingName + ConnectionStringResponseUtility.Credential);
+                    // Creating client using User assigned managed identity
+                    if (!string.IsNullOrEmpty(appSettingClientIdValue))
+                    {
+                        if (appSettingClientCredValue != ConnectionStringResponseUtility.ValidCredentialValue)
+                        {
+                            throw new ManagedIdentityException(ConnectionStringResponseUtility.ManagedIdentityCredentialMissing);
+                        }
+                        else
+                        {
+                            identityType = ConnectionStringValidationResult.ManagedIdentityType.User;
+                            client = new EventHubProducerClient(serviceUriString, eventHubName, new ManagedIdentityCredential(appSettingClientIdValue));
+                        }
+                    }
+                    // Creating client using System assigned managed identity
+                    else
+                    {
+                        identityType = ConnectionStringValidationResult.ManagedIdentityType.System;
+                        client = new EventHubProducerClient(serviceUriString, eventHubName, new ManagedIdentityCredential());
+                    }
                 }
-                else
+                catch (Exception e)
                 {
-                    response.Status = ConnectionStringValidationResult.ResultStatus.UnknownError;
+                    throw new ManagedIdentityException(e.Message, e);
```

Comment:
Still not following why we are wrapping exceptions like this instead of letting it get examined in the utility function

Reply:
this was changed in blob but not here?